### ISSUE TITLE
cn.hutool.core.io.unit.DataSizeUtil.format(java.lang.Long.MAX_VALUE) raises "java.lang.ArrayIndexOutOfBoundsException"

### ISSUE THREAD

**Issue by asdf913** [looly/hutool#1069]
### 版本情况

JDK版本：    jdk1.8.0_261 (64 bit build)
```
<dependency>
    <groupId>cn.hutool</groupId>
    <artifactId>hutool-core</artifactId>
    <version>5.4.1</version>
</dependency>
```

### 问题描述（包括截图）

1. 复现代码

```java
System.out.println(DataSizeUtil.format(Long.MAX_VALUE));
```

2. 堆栈信息
```
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 6
	at cn.hutool.core.io.unit.DataSizeUtil.format(DataSizeUtil.java:36)
```

Source 
https://github.com/looly/hutool/blob/5.4.1/hutool-core/src/main/java/cn/hutool/core/io/unit/DataSizeUtil.java

**Comment by asdf913** [comment]
Before
```
public static String format(long size) {
		if (size <= 0) {
			return "0";
		}
		int digitGroups = (int) (Math.log10(size) / Math.log10(1024));
		return new DecimalFormat("#,##0.##")
				.format(size / Math.pow(1024, digitGroups)) + " " + DataUnit.UNIT_NAMES[digitGroups];
	}
```

Suggestion
```
	public static String format(long size) {
		if (size <= 0) {
			return "0";
		}
		final Object[] unitNames = DataUnit.UNIT_NAMES;
		int digitGroups =Math.min(unitNames.length-1, (int) (Math.log10(size) / Math.log10(1024)));
		return new DecimalFormat("#,##0.##")
				.format(size / Math.pow(1024, digitGroups)) + " " + DataUnit.UNIT_NAMES[digitGroups];
	}
```

Output of suggested fix
```
8,192 EB
```

**Comment by looly** [comment]
Thanks, I will fix it at 5.4.2.

> PS: you can make pr(pull request) next time.

**Comment by asdf913** [comment]
Thanks you for your quick action!!
